unstructured_ingest/v2/processes/connectors/notion/connector.py

```diff
@@ -0,0 +1,342 @@
+import typing as t
+from dataclasses import dataclass
+from time import time
+from typing import Any, Generator, List, Optional, Set, Tuple
+from uuid import UUID
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import SourceConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+if t.TYPE_CHECKING:
+    from unstructured_ingest.v2.processes.connectors.notion.client import Client as NotionClient
+
+
+NOTION_API_VERSION = "2022-06-28"
+CONNECTOR_TYPE = "notion"
+
+
+class NotionAccessConfig(AccessConfig):
+    notion_api_key: str = Field(description="Notion API key")
+
+
+class NotionConnectionConfig(ConnectionConfig):
+    access_config: Secret[NotionAccessConfig]
+
+
+class NotionIndexerConfig(IndexerConfig):
+    page_ids: Optional[List[str]] = Field(
```

Comment:
`list` is a built in python type now.
```suggestion
    page_ids: Optional[list[str]] = Field(
```